{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "### Neovim version (nvim -v)\n\n0.7.0  compiled by runneradmin@fv-az320-113\n\n### Language server name/version\n\nclangd_14.0.3\n\n### Operating system/version\n\nwin7\n\n### Steps to reproduce using \"nvim -u minimal_init.lua\"\n\nI have gotten the reason.\r\n\r\n**1  Error**\r\n        When finding references,  failed with error:  \r\n        Error executing vim.schedule lua callback: \\nvim-win64\\share\\nvim\\runtime/lua/vim/lsp/util.lua:1616: index out of range\r\nstack traceback:\r\n\t[C]: in function '_str_byteindex_enc'\r\n\t...\\nvim-win64\\share\\nvim\\runtime/lua/vim/lsp/util.lua:1616: in function 'locations_to_items'\r\n\t...\\nvim-win64\\share\\nvim\\runtime/lua/vim/lsp/handlers.lua:192: in function 'handler'\r\n\tD:\\backup\\nvim-win64\\share\\nvim\\runtime/lua/vim/lsp.lua:1025: in function 'cb'\r\n\tvim/_editor.lua:256: in function <vim/_editor.lua:256>\r\n\r\n**2  Reason**\r\n        **Error occurs here: util.lua **\r\nfunction M._str_byteindex_enc(line, index, encoding)\r\n  if not encoding then encoding = 'utf-16' end\r\n  if encoding == 'utf-8' then\r\n    if index then return index else return #line end\r\n  elseif encoding == 'utf-16' then\r\n    return vim.str_byteindex(line, index, true)           **##########  invalid index here ######**\r\n  elseif encoding == 'utf-32' then\r\n    return vim.str_byteindex(line, index)                   **##########  or invalid index here ######**\r\n  else\r\n    error(\"Invalid encoding: \" .. vim.inspect(encoding))\r\n  end\r\nend\r\n\r\n    Because some c files are identified as CRLF format, but some lines in these files end with CR CR LF (0x0d 0d 0a). So the variable 'line' passed to _str_byteindex_enc is not the exact line,  sometime is shorter.  \n\n### Expected behavior\n\n_No response_\n\n### Actual behavior\n\n_No response_\n\n### Log file\n\n_No response_",
    "closed_at": null,
    "comment_data": [
        {
            "author_association": "MEMBER",
            "body": "Thanks for the report. Can you try the latest stable release (0.7.2), and/or the development version? \r\n\r\n- The [Releases](https://github.com/neovim/neovim/releases/tag/nightly) page has pre-built archives for Linux/Windows/macOS. \r\n- Check `:version` to confirm that you are using the latest version.",
            "created_at": "2022-07-05T13:31:52Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1175065170",
            "id": 1175065170,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5GChJS",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175065170/reactions"
            },
            "updated_at": "2022-07-05T13:32:14Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175065170",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk"
            }
        },
        {
            "author_association": "NONE",
            "body": "I tried out the latest release (0.7.2), and unfortunately, the problem is the same. ",
            "created_at": "2022-07-05T14:18:27Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1175116637",
            "id": 1175116637,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5GCttd",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175116637/reactions"
            },
            "updated_at": "2022-07-05T14:18:27Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175116637",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/45509755?v=4",
                "events_url": "https://api.github.com/users/general0000/events{/privacy}",
                "followers_url": "https://api.github.com/users/general0000/followers",
                "following_url": "https://api.github.com/users/general0000/following{/other_user}",
                "gists_url": "https://api.github.com/users/general0000/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/general0000",
                "id": 45509755,
                "login": "general0000",
                "node_id": "MDQ6VXNlcjQ1NTA5NzU1",
                "organizations_url": "https://api.github.com/users/general0000/orgs",
                "received_events_url": "https://api.github.com/users/general0000/received_events",
                "repos_url": "https://api.github.com/users/general0000/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/general0000/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/general0000/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/general0000"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Hard to read the description but it looks like you narrowed down the issue to:\r\n\r\n> Because some c files are identified as CRLF format, but some lines in these files end with CR CR LF (0x0d 0d 0a). So the variable 'line' passed to _str_byteindex_enc is not the exact line,  sometime is shorter.  ",
            "created_at": "2022-07-05T14:28:42Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1175128294",
            "id": 1175128294,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5GCwjm",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175128294/reactions"
            },
            "updated_at": "2022-07-05T14:28:42Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175128294",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1359421?v=4",
                "events_url": "https://api.github.com/users/justinmk/events{/privacy}",
                "followers_url": "https://api.github.com/users/justinmk/followers",
                "following_url": "https://api.github.com/users/justinmk/following{/other_user}",
                "gists_url": "https://api.github.com/users/justinmk/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/justinmk",
                "id": 1359421,
                "login": "justinmk",
                "node_id": "MDQ6VXNlcjEzNTk0MjE=",
                "organizations_url": "https://api.github.com/users/justinmk/orgs",
                "received_events_url": "https://api.github.com/users/justinmk/received_events",
                "repos_url": "https://api.github.com/users/justinmk/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/justinmk/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/justinmk/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/justinmk"
            }
        },
        {
            "author_association": "NONE",
            "body": "Sorry. For example,   to find reference of the function 'foo'.  When encountering such a file:\r\nline1  void bar()\r\nline2  {^M\r\nline3       foo();\r\nline4  }\r\nthe line2 ends with \\r\\r\\n ， the line number referencing foo is 3 (and column is 4), but the line processed by _str_byteindex_enc is line4.  At line 4, there is no column 4, so triggered the error  “ index out of range”。",
            "created_at": "2022-07-05T15:22:08Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1175187034",
            "id": 1175187034,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5GC-5a",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175187034/reactions"
            },
            "updated_at": "2022-07-06T00:19:31Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1175187034",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/45509755?v=4",
                "events_url": "https://api.github.com/users/general0000/events{/privacy}",
                "followers_url": "https://api.github.com/users/general0000/followers",
                "following_url": "https://api.github.com/users/general0000/following{/other_user}",
                "gists_url": "https://api.github.com/users/general0000/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/general0000",
                "id": 45509755,
                "login": "general0000",
                "node_id": "MDQ6VXNlcjQ1NTA5NzU1",
                "organizations_url": "https://api.github.com/users/general0000/orgs",
                "received_events_url": "https://api.github.com/users/general0000/received_events",
                "repos_url": "https://api.github.com/users/general0000/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/general0000/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/general0000/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/general0000"
            }
        },
        {
            "author_association": "NONE",
            "body": "[test_cr.zip](https://github.com/neovim/neovim/files/9052439/test_cr.zip)\r\n\r\nIn short, when there are mixed \\r  \\r\\n in c files, because the parsing principle of the line number between lsp client and  server is inconsistent, the client gets the wrong row content, and if the row content is shorter, the column number will be invalid,  so vim.str_byteindex produces the error: index out of range.\r\n\r\nI have confirmed this conclusion by adding logs.\r\n\r\nUsing the file in attachment can reproduce the problem. Finding any symbol references after line 7 will produce the problem.",
            "created_at": "2022-07-06T11:41:34Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1176118657",
            "id": 1176118657,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5GGiWB",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1176118657/reactions"
            },
            "updated_at": "2022-07-06T11:41:34Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1176118657",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/45509755?v=4",
                "events_url": "https://api.github.com/users/general0000/events{/privacy}",
                "followers_url": "https://api.github.com/users/general0000/followers",
                "following_url": "https://api.github.com/users/general0000/following{/other_user}",
                "gists_url": "https://api.github.com/users/general0000/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/general0000",
                "id": 45509755,
                "login": "general0000",
                "node_id": "MDQ6VXNlcjQ1NTA5NzU1",
                "organizations_url": "https://api.github.com/users/general0000/orgs",
                "received_events_url": "https://api.github.com/users/general0000/received_events",
                "repos_url": "https://api.github.com/users/general0000/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/general0000/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/general0000/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/general0000"
            }
        },
        {
            "author_association": "NONE",
            "body": "I am also experiencing the same issue with \r\n\r\nnvim 0.8.1\r\ncentos 7.9.2009\r\npython/pyright-langserver 1.1.281\r\n\r\nThe problem goes away if I just save the file in question after using the 'black' formatter.\r\n\r\nTherefore, this is not just a windows platform issue.",
            "created_at": "2022-11-28T06:25:39Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1328603991",
            "id": 1328603991,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5PMONX",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 2,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 2,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1328603991/reactions"
            },
            "updated_at": "2022-12-09T15:08:15Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1328603991",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3265849?v=4",
                "events_url": "https://api.github.com/users/MagicRedDeer/events{/privacy}",
                "followers_url": "https://api.github.com/users/MagicRedDeer/followers",
                "following_url": "https://api.github.com/users/MagicRedDeer/following{/other_user}",
                "gists_url": "https://api.github.com/users/MagicRedDeer/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/MagicRedDeer",
                "id": 3265849,
                "login": "MagicRedDeer",
                "node_id": "MDQ6VXNlcjMyNjU4NDk=",
                "organizations_url": "https://api.github.com/users/MagicRedDeer/orgs",
                "received_events_url": "https://api.github.com/users/MagicRedDeer/received_events",
                "repos_url": "https://api.github.com/users/MagicRedDeer/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/MagicRedDeer/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/MagicRedDeer/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/MagicRedDeer"
            }
        },
        {
            "author_association": "NONE",
            "body": "Also encountering this issue.\r\n\r\nnvim 0.8.1\r\nUbuntu 20.04\r\nlsp client: pyright 1.1.282",
            "created_at": "2022-12-08T02:12:20Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1341879481",
            "id": 1341879481,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5P-3S5",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1341879481/reactions"
            },
            "updated_at": "2022-12-09T15:08:21Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1341879481",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/39915562?v=4",
                "events_url": "https://api.github.com/users/leonasdev/events{/privacy}",
                "followers_url": "https://api.github.com/users/leonasdev/followers",
                "following_url": "https://api.github.com/users/leonasdev/following{/other_user}",
                "gists_url": "https://api.github.com/users/leonasdev/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/leonasdev",
                "id": 39915562,
                "login": "leonasdev",
                "node_id": "MDQ6VXNlcjM5OTE1NTYy",
                "organizations_url": "https://api.github.com/users/leonasdev/orgs",
                "received_events_url": "https://api.github.com/users/leonasdev/received_events",
                "repos_url": "https://api.github.com/users/leonasdev/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/leonasdev/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/leonasdev/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/leonasdev"
            }
        },
        {
            "author_association": "NONE",
            "body": "I'm facing the exact same thing but with cssls. When jumping to a className definition, some work other classes won't. I can jump to `bookingSummary`, but no `noShadow` or `noBorder`. I noticed that if I move one of those css declaration at the top of the file it'll work.\r\n\r\n![image](https://user-images.githubusercontent.com/3494708/227153302-416870c8-74f8-4520-9460-0845c1adacd6.png)\r\n",
            "created_at": "2023-03-23T09:00:30Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1480819036",
            "id": 1480819036,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5YQ4Fc",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1480819036/reactions"
            },
            "updated_at": "2023-03-23T09:04:21Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1480819036",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3494708?v=4",
                "events_url": "https://api.github.com/users/vricop/events{/privacy}",
                "followers_url": "https://api.github.com/users/vricop/followers",
                "following_url": "https://api.github.com/users/vricop/following{/other_user}",
                "gists_url": "https://api.github.com/users/vricop/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/vricop",
                "id": 3494708,
                "login": "vricop",
                "node_id": "MDQ6VXNlcjM0OTQ3MDg=",
                "organizations_url": "https://api.github.com/users/vricop/orgs",
                "received_events_url": "https://api.github.com/users/vricop/received_events",
                "repos_url": "https://api.github.com/users/vricop/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/vricop/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/vricop/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/vricop"
            }
        },
        {
            "author_association": "NONE",
            "body": "I am also facing the same issue with ccls. For some symbols navigation works for some it doesn't.\r\n![Screenshot from 2023-04-12 09-34-49](https://user-images.githubusercontent.com/22386546/231346854-c19714f9-82c0-474d-85f7-360a8f2b5090.png)\r\n",
            "created_at": "2023-04-12T04:07:20Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1504566626",
            "id": 1504566626,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5Zrd1i",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1504566626/reactions"
            },
            "updated_at": "2023-06-05T02:03:29Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1504566626",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/22386546?v=4",
                "events_url": "https://api.github.com/users/jbarik/events{/privacy}",
                "followers_url": "https://api.github.com/users/jbarik/followers",
                "following_url": "https://api.github.com/users/jbarik/following{/other_user}",
                "gists_url": "https://api.github.com/users/jbarik/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/jbarik",
                "id": 22386546,
                "login": "jbarik",
                "node_id": "MDQ6VXNlcjIyMzg2NTQ2",
                "organizations_url": "https://api.github.com/users/jbarik/orgs",
                "received_events_url": "https://api.github.com/users/jbarik/received_events",
                "repos_url": "https://api.github.com/users/jbarik/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/jbarik/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/jbarik/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/jbarik"
            }
        },
        {
            "author_association": "NONE",
            "body": "@jbarik I found out what's causing this. You just need to install the cssmodule_ls server, I'm myself using Mason for that. Then you need to pass some specific server settings so it doesn't conflict with 'Go to definition' from the tsserver.\r\n\r\n![image](https://user-images.githubusercontent.com/3494708/232250853-58d71c9d-f49f-4920-aed2-a21429861cdc.png)\r\n\r\nSo this doesn't have to do with native Neovim LSP.\r\n\r\nHere's my config[^*]:\r\n\r\n[^*]: Do the same thing if your using just lspconfig.\r\n\r\n```lua\r\nrequire('mason-lspconfig').setup_handlers({\r\n  function(server_name)\r\n    lspconfig[server_name].setup({\r\n      on_attach = on_attach,\r\n      capabilities = capabilities,\r\n      settings = {\r\n        Lua = {\r\n          diagnostics = { globals = { 'vim' } },\r\n        },\r\n        -- NOTE: For some reason, stylelint isn't loading my config file with \r\n        -- rules for avoiding false positives when using PostCss plugins.\r\n        css = {\r\n          validate = true,\r\n          lint = {\r\n            unknownAtRules = 'ignore',\r\n          },\r\n        },\r\n        cssmodules = { -- <= This is what you need, don't forget to install the server\r\n          capabilities = {\r\n            definitionProvider = false\r\n          }\r\n        }\r\n      },\r\n    })\r\n  end,\r\n})\r\n```\r\n\r\nSource: https://github.com/antonk52/cssmodules-language-server#configuration\r\n\r\nI hope it helps...",
            "created_at": "2023-04-15T19:58:19Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1509945822",
            "id": 1509945822,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5Z__He",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1509945822/reactions"
            },
            "updated_at": "2023-04-16T20:22:28Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1509945822",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/3494708?v=4",
                "events_url": "https://api.github.com/users/vricop/events{/privacy}",
                "followers_url": "https://api.github.com/users/vricop/followers",
                "following_url": "https://api.github.com/users/vricop/following{/other_user}",
                "gists_url": "https://api.github.com/users/vricop/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/vricop",
                "id": 3494708,
                "login": "vricop",
                "node_id": "MDQ6VXNlcjM0OTQ3MDg=",
                "organizations_url": "https://api.github.com/users/vricop/orgs",
                "received_events_url": "https://api.github.com/users/vricop/received_events",
                "repos_url": "https://api.github.com/users/vricop/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/vricop/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/vricop/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/vricop"
            }
        },
        {
            "author_association": "NONE",
            "body": "That doesn't fix the issue for `tsserver`, though. I use TypeScript and can confirm that the original issue still exists. Though, from what I can tell, I don't have mixed line feeds. Maybe my issue is not related to this issue after all 🧐 ",
            "created_at": "2023-04-18T10:12:13Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1512812626",
            "id": 1512812626,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5aK7BS",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1512812626/reactions"
            },
            "updated_at": "2023-04-18T10:12:13Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1512812626",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/1196244?v=4",
                "events_url": "https://api.github.com/users/schemar/events{/privacy}",
                "followers_url": "https://api.github.com/users/schemar/followers",
                "following_url": "https://api.github.com/users/schemar/following{/other_user}",
                "gists_url": "https://api.github.com/users/schemar/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/schemar",
                "id": 1196244,
                "login": "schemar",
                "node_id": "MDQ6VXNlcjExOTYyNDQ=",
                "organizations_url": "https://api.github.com/users/schemar/orgs",
                "received_events_url": "https://api.github.com/users/schemar/received_events",
                "repos_url": "https://api.github.com/users/schemar/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/schemar/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/schemar/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/schemar"
            }
        },
        {
            "author_association": "NONE",
            "body": "As a heads up, I also ran into this error, but it was due to a bug in my plugin's implementation. I was trying to load a custom scheme in the form of `distant://1234@/path/to/file`, but the buffer was returning empty even though the LSP provided the correct line & column position. When jumping to a definition, it tried to load the file, but that yielded an empty buffer. This resulted in a call to `_str_byteindex_enc` with an empty line while still being provided a valid encoding (UTF-16) and index.\r\n\r\nMight be good to catch a disconnect where it's clearly going to error and report a more meaningful error on top of the index issue.",
            "created_at": "2023-05-23T19:04:46Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1559985000",
            "id": 1559985000,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5c-3to",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 3,
                "-1": 0,
                "confused": 0,
                "eyes": 1,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 4,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1559985000/reactions"
            },
            "updated_at": "2023-05-23T19:04:46Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1559985000",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/2481802?v=4",
                "events_url": "https://api.github.com/users/chipsenkbeil/events{/privacy}",
                "followers_url": "https://api.github.com/users/chipsenkbeil/followers",
                "following_url": "https://api.github.com/users/chipsenkbeil/following{/other_user}",
                "gists_url": "https://api.github.com/users/chipsenkbeil/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/chipsenkbeil",
                "id": 2481802,
                "login": "chipsenkbeil",
                "node_id": "MDQ6VXNlcjI0ODE4MDI=",
                "organizations_url": "https://api.github.com/users/chipsenkbeil/orgs",
                "received_events_url": "https://api.github.com/users/chipsenkbeil/received_events",
                "repos_url": "https://api.github.com/users/chipsenkbeil/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/chipsenkbeil/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/chipsenkbeil/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/chipsenkbeil"
            }
        },
        {
            "author_association": "NONE",
            "body": "I debugged this issue a little more and this is what I found:\r\n\r\nWhen I try to jump to the header file from a function definition in cpp file, I intercepted the results from ccls(at \r\n **/usr/share/nvim/runtime/lua/vim/lsp/handlers.lua, function location_handler**). The result from ccls contained multiple targetUri, namely:\r\n\r\n```\r\ntargetUri = \"file://../../../../../../src/sl_lang_blocks/export/include/sl_lang_blocks/ControlPortBlock.hpp\" \r\ntargetUri = \"file://../../../src/sl_lang_blocks/export/include/sl_lang_blocks/ControlPortBlock.hpp\"\r\ntargetUri = \"file://../../src/sl_lang_blocks/export/include/sl_lang_blocks/ControlPortBlock.hpp\"\r\ntargetUri = \"file://../../src/sl_lang_blocks/export/include/sl_lang_blocks/ControlPortBlock.hpp\"\r\ntargetUri = \"file:///local-ssd/jbarik/ls_eventport2/matlab/src/sl_lang_blocks/export/include/sl_lang_blocks/ControlPortBlock.hpp\"\r\n```\r\n\r\nI was expecting only one, that is the last one which contains the absolute path. Even if I consider the relative path of the hpp from the location of the cpp file, of all the targetUri, one is correct.\r\n\r\nLater, in **function M.locations_to_items** (/usr/share/nvim/runtime/lua/vim/lsp/util.lua), some of those Uri which contain relative path(file://../../***) fail to resolve to a proper file and the code ends up sending an empty line to **M._str_byteindex_enc** which errors out.\r\n\r\n\r\n\r\n                                                     \r\n",
            "created_at": "2023-07-19T13:40:48Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1642106710",
            "id": 1642106710,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5h4I9W",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1642106710/reactions"
            },
            "updated_at": "2023-07-19T13:42:36Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1642106710",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/22386546?v=4",
                "events_url": "https://api.github.com/users/jbarik/events{/privacy}",
                "followers_url": "https://api.github.com/users/jbarik/followers",
                "following_url": "https://api.github.com/users/jbarik/following{/other_user}",
                "gists_url": "https://api.github.com/users/jbarik/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/jbarik",
                "id": 22386546,
                "login": "jbarik",
                "node_id": "MDQ6VXNlcjIyMzg2NTQ2",
                "organizations_url": "https://api.github.com/users/jbarik/orgs",
                "received_events_url": "https://api.github.com/users/jbarik/received_events",
                "repos_url": "https://api.github.com/users/jbarik/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/jbarik/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/jbarik/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/jbarik"
            }
        },
        {
            "author_association": "NONE",
            "body": "Facing the same issue invoking `lua vim.lsp.buf.references()` with clangd running on remote index:\r\n\r\n```\r\nError executing vim.schedule lua callback: ...lar/neovim/0.9.4/share/nvim/runtime/lua/vim/lsp/util.lua:1833: index out of range\r\nstack traceback:\r\n\t[C]: in function '_str_byteindex_enc'\r\n\t...lar/neovim/0.9.4/share/nvim/runtime/lua/vim/lsp/util.lua:1833: in function 'locations_to_items'\r\n\t...nvim/lazy/telescope.nvim/lua/telescope/builtin/__lsp.lua:27: in function 'handler'\r\n\t...l/Cellar/neovim/0.9.4/share/nvim/runtime/lua/vim/lsp.lua:1393: in function ''\r\n\tvim/_editor.lua: in function <vim/_editor.lua:0>\r\n\r\n```\r\n\r\nAdding debug prints at `/usr/local/Cellar/neovim/0.9.4/share/nvim/runtime/lua/vim/lsp/util.lua` before the `_str_byteindex_enc` invocation produces the following before raising an error:\r\n\r\n\r\n```\r\n{ \"  // The process should crash when going through abnormal termination, but we\", 2, \"utf-16\" }\r\n{ \"    std::size(kNonWildcardDomainNonPortSchemes);\", 18, \"utf-16\" }\r\n{ \"\", 13, \"utf-16\" }\r\n```\r\n\r\nThe issue seems to be `pos.character` being evaluated against an empty line.\r\n\r\n> The issue is the divergent remote index producing out-of-line references that do not exist in the local codebase\r\n\r\n## Hotfix\r\n\r\n```\r\nlocal ok, col = pcall(M._str_byteindex_enc, line, pos.character, offset_encoding)\r\nif ok then\r\n  table.insert...\r\n```\r\n",
            "created_at": "2024-01-09T04:42:53Z",
            "html_url": "https://github.com/neovim/neovim/issues/19237#issuecomment-1882404055",
            "id": 1882404055,
            "issue_url": "https://api.github.com/repos/neovim/neovim/issues/19237",
            "node_id": "IC_kwDOAPphoM5wMzTX",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 2,
                "-1": 0,
                "confused": 0,
                "eyes": 1,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 3,
                "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1882404055/reactions"
            },
            "updated_at": "2024-01-09T04:48:56Z",
            "url": "https://api.github.com/repos/neovim/neovim/issues/comments/1882404055",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/59549574?v=4",
                "events_url": "https://api.github.com/users/teocns/events{/privacy}",
                "followers_url": "https://api.github.com/users/teocns/followers",
                "following_url": "https://api.github.com/users/teocns/following{/other_user}",
                "gists_url": "https://api.github.com/users/teocns/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/teocns",
                "id": 59549574,
                "login": "teocns",
                "node_id": "MDQ6VXNlcjU5NTQ5NTc0",
                "organizations_url": "https://api.github.com/users/teocns/orgs",
                "received_events_url": "https://api.github.com/users/teocns/received_events",
                "repos_url": "https://api.github.com/users/teocns/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/teocns/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/teocns/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/teocns"
            }
        }
    ],
    "comments": 14,
    "comments_url": "https://api.github.com/repos/neovim/neovim/issues/19237/comments",
    "created_at": "2022-07-05T13:13:29Z",
    "events_url": "https://api.github.com/repos/neovim/neovim/issues/19237/events",
    "html_url": "https://github.com/neovim/neovim/issues/19237",
    "id": 1294285168,
    "labels": [
        {
            "color": "f9d0c4",
            "default": true,
            "description": "issues reporting wrong behavior",
            "id": 77997474,
            "name": "bug",
            "node_id": "MDU6TGFiZWw3Nzk5NzQ3NA==",
            "url": "https://api.github.com/repos/neovim/neovim/labels/bug"
        },
        {
            "color": "c5def5",
            "default": false,
            "description": null,
            "id": 662566370,
            "name": "lsp",
            "node_id": "MDU6TGFiZWw2NjI1NjYzNzA=",
            "url": "https://api.github.com/repos/neovim/neovim/labels/lsp"
        }
    ],
    "labels_url": "https://api.github.com/repos/neovim/neovim/issues/19237/labels{/name}",
    "locked": false,
    "milestone": null,
    "node_id": "I_kwDOAPphoM5NJTlw",
    "number": 19237,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 5,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 5,
        "url": "https://api.github.com/repos/neovim/neovim/issues/19237/reactions"
    },
    "repository_url": "https://api.github.com/repos/neovim/neovim",
    "state": "open",
    "state_reason": null,
    "timeline_url": "https://api.github.com/repos/neovim/neovim/issues/19237/timeline",
    "title": "LSP: finding references \"index out of range\" with mixed LF, CRLF files",
    "updated_at": "2024-01-09T04:48:56Z",
    "url": "https://api.github.com/repos/neovim/neovim/issues/19237",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/45509755?v=4",
        "events_url": "https://api.github.com/users/general0000/events{/privacy}",
        "followers_url": "https://api.github.com/users/general0000/followers",
        "following_url": "https://api.github.com/users/general0000/following{/other_user}",
        "gists_url": "https://api.github.com/users/general0000/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/general0000",
        "id": 45509755,
        "login": "general0000",
        "node_id": "MDQ6VXNlcjQ1NTA5NzU1",
        "organizations_url": "https://api.github.com/users/general0000/orgs",
        "received_events_url": "https://api.github.com/users/general0000/received_events",
        "repos_url": "https://api.github.com/users/general0000/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/general0000/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/general0000/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/general0000"
    }
}